airflow/models/dagbag.py

```diff
@@ -510,6 +513,6 @@ def sync_to_db(self):
         DAG.bulk_sync_to_db(self.dags.values())
         # Write Serialized DAGs to DB if DAG Serialization is turned on
         # Even though self.read_dags_from_db is False
-        if settings.STORE_SERIALIZED_DAGS:
+        if settings.STORE_SERIALIZED_DAGS or self.read_dags_from_db:
```

Comment:
Can you tell a little more about it? How do you want serialization of the same object to a database to produce different results?  Is it by adding the `next_dagrun`, `next_dagrun_create_after` fields to the DAG in the sync_to_db method?

Reply:
@ashb can you please take a look here?

Reply:
This change was a bit of a hack here, and we'll tidy it up.

Since the scheduler now requires DAG serialization to operate, we need a way for dag bags to Load dags from given files only, and not from the DB, but then also a way to make them _write_ things back to the DB.

We'll pull this out in to a separate PR, essentially removing the `STORE_SERIALIZED_DAGS` option entirely, and then this bulk_sync_to_db can then just always write to the DB.